### DISCUSSION TITLE
error: ‘btintptr_t’ was not declared in this scope for BUILD_EXTRAS

### DISCUSSION THREAD

**Discussion by jiapei100** [bulletphysics/bullet3#3526] [General]
The Error...

`/bullet3/Extras/Serialize/makesdna/makesdna.cpp:1068:9: error: ‘btintptr_t’ was not declared in this scope
`

**Comment by erwincoumans** [comment]
It compiles fine on Windows, Linux and Mac OSX, so please provide ALL details, including operating system+version, compiler+version, all build system steps etc.

**Comment by benelot** [comment]
Just had this on Linux as well. I could compile it recently, but not anymore since I upgraded TF. I tried to remember the changes, but could not find the reason. Is 'btintptr_t' defined somewhere for Linux? I could not find any reference in the code where it would be defined in the linux case...

**Comment by iSLC** [comment]
If your compiler does not define `_INTPTR_T_DEFINED` and neither declares `intptr_t` somewhere (_c++11 not enabled?_) then declaring  `btintptr_t`  as `intptr_t` is equal to `btintptr_t` not being declared. 

I suppose that macro condition can be written as:
```
#if defined(INTPTR_MIN) || defined(INTPTR_MAX)
	typedef intptr_t btintptr_t;
#elif defined(_WIN64)
	typedef __int64 btintptr_t;
#else
	typedef long btintptr_t;
#endif
```

Although I don't see `stdint.h` included anywhere in there.

**EDIT**: nvm, `stdint.h` is included but is after that macro. which probably causes the mess.

**EDIT2**: Upon closer look, `btintptr_t` seems to be declared only on windows in the `_WIN32` branch of the macros. Lack of indentation made it a bit harder to notice.

**Comment by nbelakovski** [comment]
I just had this same issue compiling makesDNA on mac:
/Users/nickolai.belakovski/develop/bullet3/Extras/Serialize/makesdna/makesdna.cpp:1074:9: error: use of undeclared identifier 'btintptr_t'
It looks like the #defines that have been referenced in this thread resolve to #include <inttypes.h> for Mac. Adding `typedef long btintptr_t` right after that include fixed the compile issue, not sure if that's the desired fix though.

**Comment by erwincoumans** [comment]
Why/how would you compile makesdna.cpp? The file is optional and shouldn't be compiled by default. Did you use cmake and set INTERNAL_UPDATE_SERIALIZATION_STRUCTURES to ON (it is OFF) by default?
If so, can you just turn it OFF again?

**Comment by nbelakovski** [comment]
I'm trying to extend the serialization structures, so, yea, I am compiling this on purpose.
